File: ADT/graphs/bfs.js
function bfs(adj){
    const V = adj.length
    const visited = new Array(V).fill(false)
    const res = []
    const q = new Denque()
    let src = 0
    visited[src] = true
    q.push(src)

    while(!q.isEmpty()){
        const curr = q.shift()
        res.push(curr)

        for(const x of adj[curr]){
            if(!visited[x]){
                visited[x] = true
                q.push(x)
            }
        }
    }
    return res
}

File: ADT/graphs/bfs.py
from collections import deque
def bfs(adj):
    V = len(adj)
    visited = [False] * V
    res = []
    src = 0
    q = deque()
    visited[src] = True
    q.append(src)

    while q:
        curr = q.popleft()
        res.append(curr)

        for x in adj[curr]:
            if not visited[x]:
                visited[x] = True
                q.append(x)

    return res